packages/babel-plugin-extract-export-names/test/index.test.js

```diff
@@ -2,55 +2,137 @@ const babel = require('@babel/core')
 
 const BabelPluginExtractExportNames = require('..')
 
-const FIXTURE = `
-export const foo = 'bar'
-export const foo2 = {
-  bar: 'baze',
-  hi: \`Hello! \${foo.toJSON() || 'hrm'}\`,
-  abc: {
-    def: 123
-  }
-}
-export const foo3 = [1, 2, 3, { a: 'b' }]
-export const A = 'baz'
-export const [B] = [1]
-export const [C, D, E] = [a, { b: 'c' }]
-export const { F } = { foo: 'bar' }
-export const { G, H } = {}
-export { Super } from './super'
-`
-
-const transform = str => {
+const transform = value => {
   const plugin = new BabelPluginExtractExportNames()
 
-  const result = babel.transform(str, {
-    configFile: false,
-    plugins: [plugin.plugin]
-  })
+  babel.transformSync(value, {configFile: false, plugins: [plugin.plugin]})
 
-  return {
-    ...result,
-    state: plugin.state
-  }
+  return plugin.state.names
 }
 
 describe('babel-plugin-extract-export-names', () => {
-  test('adds export names to state', () => {
-    const result = transform(FIXTURE)
-
-    expect(result.state.names).toEqual([
-      'foo',
-      'foo2',
-      'foo3',
-      'A',
-      'B',
-      'C',
-      'D',
-      'E',
-      'F',
-      'G',
-      'H',
-      'Super'
+  test('should capture an empty list on an empty file', () => {
+    expect(transform('')).toEqual([])
+  })
+
+  test('should capture an empty list if nothing is exported', () => {
+    expect(transform('console.log(1)')).toEqual([])
+  })
+
+  test('should capture `const`, `let`, and `var`', () => {
+    const list = ['var', 'let', 'const']
+    const expected = ['name1', 'name2', 'name3']
+    const input = list
+      .map((d, i) => `export ${d} ${expected[i]} = ""`)
+      .join('\n')
+    expect(transform(input)).toEqual(expected)
+  })
+
+  test('should capture multiple exports with commas', () => {
+    expect(
+      transform('var name1 = "a", name2 = "b"; export var name1, name2')
+    ).toEqual(['name1', 'name2'])
+  })
+
+  test('should capture multiple exports with commas and assignments', () => {
+    expect(transform('export var name1 = "a", name2 = "b"')).toEqual([
+      'name1',
+      'name2'
+    ])
+  })
+
+  // To do: should be added.
+  test('should *not* capture a named function', () => {
+    expect(transform('export function name1() {}')).toEqual([])
+  })
+
+  // To do: should be added.
+  test('should *not* capture a class', () => {
+    expect(transform('export class name1 {}')).toEqual([])
+  })
+
+  test('should capture a list', () => {
+    expect(
+      transform('var name1 = "a", name2 = "b"; export {name1, name2}')
+    ).toEqual(['name1', 'name2'])
+  })
+
+  // To do: is this correct?
+  test('should capture renames', () => {
+    expect(
+      transform(
+        'var name1 = "a", name2 = "b"; export {name1 as foo, name2 as bar}'
+      )
+    ).toEqual(['name1', 'name2'])
+  })
```

Comment:
Good catch.
The names should be aliased as `foo` and `bar`

Reply:
I guess this package is used more for figuring out which local values are also exported (not caring about as what they’re exported).

https://github.com/mdx-js/mdx/blob/e194fc1a61715549be10d56055ea4ad25533fcb1/packages/mdx/mdx-hast-to-jsx.js#L153-L163

I’m not 100% whether doing it like this, and at build time, is useful? I’d say checking if this was nully at runtime would be better?

https://github.com/mdx-js/mdx/blob/9443e235679f7d37cdfa436ef1b167e4bb68615d/packages/react/src/create-element.js#L23-L27

Reply:
I think I’ll just leave {import,export}-names and see when I start on mdx-js/mdx whether they can be deprecated